src/Strategy/ManifestStrategy.php

```diff
@@ -0,0 +1,178 @@
+<?php
+namespace Humbug\SelfUpdate\Strategy;
+
+use Humbug\SelfUpdate\Exception\HttpRequestException;
+use Humbug\SelfUpdate\Exception\JsonParsingException;
+use Humbug\SelfUpdate\Updater;
+use Humbug\SelfUpdate\VersionParser;
+
+class ManifestStrategy implements StrategyInterface
+{
+    /** @var array */
+    private static $requiredKeys = array('sha1', 'version', 'url');
+    /** @var string */
+    private $manifestUrl;
+    /** @var array */
+    private $manifest;
+    /** @var array */
+    private $availableVersions;
+    /** @var string */
+    private $localVersion;
+    /** @var bool */
+    private $allowMajor = false;
+    /** @var bool */
+    private $allowUnstable = false;
+
+    /**
+     * ManifestStrategy constructor.
```

Comment:
I'm personally not very found of noisy comments (i.e. comment not providing any value), so I would remove that line unless @padraic prefer to keep it for consistency. Although I would argue in favour of changing that practice and fixing that inconsistency later :)